sklearn_numba_dpex/kmeans/drivers.py

```diff
@@ -416,6 +419,194 @@ def __call__(
+            dtype=dtype,
+        )
+
+        half_l2_norm = make_half_l2_norm_2d_axis0_kernel(
+            size0=n_features,
+            size1=n_clusters,
+            work_group_size=work_group_size,
+            dtype=dtype,
+        )
+
+        X_t, centroids_t = self._load_transposed_data_to_device(X, centers)
+
+        centroids_half_l2_norm = dpctl.tensor.empty(
+            n_clusters, dtype=dtype, device=self.device
+        )
+        assignments_idx = dpctl.tensor.empty(
+            n_samples, dtype=np.uint32, device=self.device
+        )
+
+        half_l2_norm(centroids_t, centroids_half_l2_norm)
+
+        label_assignment_fixed_window_kernel(
+            X_t,
+            centroids_t,
+            centroids_half_l2_norm,
+            assignments_idx,
+        )
+
+        return (dpctl.tensor.asnumpy(assignments_idx).astype(np.int32), None)
+
+    def get_inertia(
+        self,
+        X,
+        sample_weight,
+        x_squared_norms,
+        centers,
+        n_threads=1,
+        return_inertia=True,
+    ):
+        """This call is expected to accept the same inputs than sklearn's private
+        _labels_inertia and returns (None, inertia)
+        """
+        (
+            X,
+            centers,
+            dtype,
+            work_group_size,
+            n_features,
+            n_samples,
+            n_clusters,
+        ) = self._check_inputs(X, centers)
+
+        # Create a set of kernels
+        compute_inertia_fixed_window_kernel = make_compute_inertia_fixed_window_kernel(
+            n_samples,
+            n_features,
+            n_clusters,
+            preferred_work_group_size_multiple=self.preferred_work_group_size_multiple,
+            centroids_window_width_multiplier=self.centroids_window_width_multiplier,
+            centroids_window_height=self.centroids_window_height,
+            work_group_size=work_group_size,
+            dtype=dtype,
+        )
+
+        half_l2_norm = make_half_l2_norm_2d_axis0_kernel(
+            size0=n_features,
+            size1=n_clusters,
+            work_group_size=work_group_size,
+            dtype=dtype,
+        )
+
+        reduce_inertia = make_sum_reduction_1d_kernel(
+            size=n_samples,
+            work_group_size=work_group_size,
+            device=self.device,
+            dtype=dtype,
+        )
+
+        X_t, centroids_t = self._load_transposed_data_to_device(X, centers)
+
+        centroids_half_l2_norm = dpctl.tensor.empty(
+            n_clusters, dtype=dtype, device=self.device
+        )
+        per_sample_inertia = dpctl.tensor.empty(
+            n_samples, dtype=dtype, device=self.device
+        )
+
+        half_l2_norm(centroids_t, centroids_half_l2_norm)
+
+        compute_inertia_fixed_window_kernel(
+            X_t,
+            centroids_t,
+            centroids_half_l2_norm,
+            per_sample_inertia,
+        )
+
+        inertia = dpctl.tensor.asnumpy(reduce_inertia(per_sample_inertia))[0]
```

Comment:
Why do we need to take the first returned element solely here?
What is the instruction returning?
```suggestion
        inertia = dpctl.tensor.asnumpy(reduce_inertia_kernel(per_sample_inertia))[0]
```

Reply:
no actually I'm wrong: the current code is good, `reduce_inertia_kernel(per_sample_inertia)` is a 1-sized array

Reply:
In this case, I think having a comment indicates that it is a 1-sized array whose scalar is accessed via `[0]` would be useful for readers.